### PR TITLE
Implement PTP reboot-recovery test

### PR DESCRIPTION
**Pull request by bnshr**
This PR includes an automated test that makes sure PTP recovers within a reasonable timeframe after reboot.

### CODE REVIEW

File: test/conformance/ptp/ptp.go [hunk 1]
```diff
@@ -123,6 +125,81 @@ var _ = Describe("[ptp]", func() {
 
 		})
 
+		Context("PTP Reboot discovery", func() {
+			BeforeEach(func() {
+				if isSingleNode {
```
Comment (edcdavid) [inline test/conformance/ptp/ptp.go:130]: Should also work with single node

File: test/conformance/ptp/ptp.go [hunk 2]
```diff
@@ -123,6 +125,81 @@ var _ = Describe("[ptp]", func() {
...
+					Skip("Running in single node mode")
+				}
+				if discoveryFailed {
```
Comment (edcdavid) [inline test/conformance/ptp/ptp.go:133]: us this instead:  testconfig.DiscoveryFailureStatus

File: test/conformance/ptp/ptp.go [hunk 3]
```diff
@@ -943,3 +1045,294 @@ func ptpEventEnabled() bool {
...
+
+	// Find the ptp in slave machine
+	ptpSlaveRunningPods := []v1core.Pod{}
+	ptpMasterRunningPods := []v1core.Pod{}
+
+	var ptpRunningPods []v1core.Pod
+
+	ptpPodsAfterReboot, err = client.Client.Pods(PtpLinuxDaemonNamespace).List(context.Background(), metav1.ListOptions{LabelSelector: "app=linuxptp-daemon"})
+
+	for _, pod := range ptpPodsAfterReboot.Items {
```
Comment (edcdavid) [inline test/conformance/ptp/ptp.go:1278]: Should include the actual test after reboot in this funtion? or just call the reboot test before the other tests? I assume we woudl like to run reboot for the other clock types as well

**Comment by aneeshkp** [comment]
@bnshr any updates

**Comment by bnshr** [comment]
@aneeshkp Could you please review this PR?

File: test/conformance/ptp/ptp.go [hunk 4]
```diff
@@ -943,3 +989,124 @@ func ptpEventEnabled() bool {
...
+	Eventually(func() error {
+		_, err := pods.ExecCommand(client.Client, pod, "container-00", []string{"chroot", "/host", "shutdown", "-r"})
+		return err
+	}, timeout, pollingInterval).Should(BeNil())
+}
+
+func rebootNode(nodesCount int32, fullConfig testconfig.TestConfig) {
+
+	logrus.Info("Rebooting system starts ..............")
+	daemonSetName := "ptp-reboot"
```
Comment (edcdavid) [inline test/conformance/ptp/ptp.go:1075]: make these variables const

File: test/conformance/ptp/ptp.go [hunk 5]
```diff
@@ -943,3 +989,124 @@ func ptpEventEnabled() bool {
...
+	// 2. Reboot the machines
+	for _, pod := range runningPods.Items {
+		checkRestart(pod)
+	}
+	// Get all nodes
+	nodes, err := client.Client.CoreV1().Nodes().List(context.Background(), metav1.ListOptions{})
+	if err != nil {
+		panic(err.Error())
+	}
+	// Check that all nodes are not unreachable
```
Comment (edcdavid) [inline test/conformance/ptp/ptp.go:1095]: // Check that all nodes are not unreachable -> // Wait for all nodes to be unreachable

File: test/conformance/ptp/ptp.go [hunk 6]
```diff
@@ -943,3 +989,124 @@ func ptpEventEnabled() bool {
 	}
 	return ptpConfig.Spec.EventConfig.EnableEventPublisher
 }
+
+func checkAndGetPtpRunningPods(fullConfig testconfig.TestConfig) []v1core.Pod {
+	ptpPods, err := client.Client.CoreV1().Pods(PtpLinuxDaemonNamespace).List(context.Background(), metav1.ListOptions{LabelSelector: "app=linuxptp-daemon"})
+	Expect(err).NotTo(HaveOccurred())
+	Expect(len(ptpPods.Items)).To(BeNumerically(">", 0), "linuxptp-daemon is not deployed on cluster")
```
Comment (edcdavid) [inline test/conformance/ptp/ptp.go:996]: Shouldn't we have a time wait loop here to wait after nodes have been pinging successfully and before the oc command is available

File: test/conformance/ptp/ptp.go (hunk 4 above)
Comment (bnshr) [reply test/conformance/ptp/ptp.go:1075]: Done!

File: test/conformance/ptp/ptp.go (hunk 6 above)
Comment (bnshr) [reply test/conformance/ptp/ptp.go:996]: This is done in the function `rebootNode` in the code 

```java
// Wait for all nodes to be reachable now after the restart
	const reachable = true
	for _, node := range nodes.Items {
		testnode.WaitForNodeReachability(&node, 10*time.Minute, reachable)
	}`

File: test/conformance/ptp/ptp.go [hunk 7]
```diff
@@ -123,6 +127,47 @@ var _ = Describe("[ptp]", func() {
...
+
+			BeforeEach(func() {
+				if fullConfig.Status == testconfig.DiscoveryFailureStatus {
+					Skip("Failed to find a valid ptp slave configuration")
+				}
+				ptpPods, err := client.Client.CoreV1().Pods(PtpLinuxDaemonNamespace).List(context.Background(), metav1.ListOptions{LabelSelector: "app=linuxptp-daemon"})
+				Expect(err).NotTo(HaveOccurred())
+				Expect(len(ptpPods.Items)).To(BeNumerically(">", 0), "linuxptp-daemon is not deployed on cluster")
+
+				ptpSlaveRunningPods := []v1core.Pod{}
```
Comment (aneeshkp) [inline test/conformance/ptp/ptp.go:140]: Don't you already have this pod configured? Why would you want to rediscover this artifacts ?

**Pull request closed without merging**